api/src/test/java/org/openmrs/api/OrderServiceTest.java

```diff
@@ -3209,4 +3223,24 @@ public void saveRetrospectiveOrder_shouldFailIfAnActiveDrugOrderForTheSameConcep
         expectedException.expectMessage("Cannot have more than one active order for the same orderable and care setting");
         orderService.saveRetrospectiveOrder(order, null);
     }
+	@Test
+	@Verifies(value = "should not save order if invalid orderGroup encounter", method = "saveOrder(Order)")
+	public void saveOrder_shouldNotSaveOrderIfInvalidOrderGroupEncounter() throws Exception {
+		executeDataSet(ORDER_SET);
+		OrderGroup orderGroup = new OrderGroup();
+
+		final Order order = new OrderBuilder().withAction(Order.Action.NEW).withPatient(7).withConcept(1000)
+				.withCareSetting(1).withOrderer(1).withEncounter(3).withDateActivated(new Date()).withOrderType(17)
+				.withUrgency(Order.Urgency.ON_SCHEDULED_DATE).withScheduledDate(new Date()).build();
+
+		orderGroup.setEncounter(Context.getEncounterService().getEncounter(5));
+		orderGroup.setOrderSet(Context.getOrderSetService().getOrderSetById(2000));
+
+		order.setOrderGroup(orderGroup);
+
+		expectedException.expect(ValidationException.class);
```

Comment:
Would be nice to test the error message too otherwise this could be an validation exception for something else